web/src/components/tasks/business-formation/validatedFieldsForUser.test.ts

```diff
@@ -47,163 +47,276 @@ describe("validatedFieldsForUser", () => {
+    });
+  });
+
+  describe("LP", () => {
+    const legalStructureId = "limited-partnership";
+
+    it("also requires LP fields for NO answers to radio questions", () => {
+      const formationFormData = generateFormationFormData(
+        {
+          canCreateLimitedPartner: false,
+          canGetDistribution: false,
+          canMakeDistribution: false,
+        },
+        { legalStructureId }
+      );
+
+      const expected: FormationFields[] = [
+        ...validatedFieldsForAllLegalStructures,
+        "withdrawals",
+        "combinedInvestment",
+        "dissolution",
+        "canCreateLimitedPartner",
+        "canGetDistribution",
+        "canMakeDistribution",
+        "incorporators",
+      ];
+      expect(validatedFieldsForUser(formationFormData)).toEqual(expect.arrayContaining(expected));
+    });
+
+    it("also terms fields for YES answer to canCreateLimitedPartner", () => {
+      const formationFormData = generateFormationFormData(
+        {
+          legalType: "limited-partnership",
+          canCreateLimitedPartner: true,
+          canGetDistribution: false,
+          canMakeDistribution: false,
+        },
+        { legalStructureId }
+      );
+
+      expect(validatedFieldsForUser(formationFormData)).toContain("createLimitedPartnerTerms");
+      expect(validatedFieldsForUser(formationFormData)).not.toContain("getDistributionTerms");
+      expect(validatedFieldsForUser(formationFormData)).not.toContain("makeDistributionTerms");
+    });
+
+    it("also terms fields for YES answer to canGetDistribution", () => {
+      const formationFormData = generateFormationFormData(
+        {
+          canCreateLimitedPartner: true,
+          canGetDistribution: true,
+          canMakeDistribution: false,
+        },
+        { legalStructureId }
+      );
+
+      expect(validatedFieldsForUser(formationFormData)).toContain("createLimitedPartnerTerms");
+      expect(validatedFieldsForUser(formationFormData)).toContain("getDistributionTerms");
+      expect(validatedFieldsForUser(formationFormData)).not.toContain("makeDistributionTerms");
+    });
+
+    it("also terms fields for YES answer to canMakeDistribution", () => {
+      const formationFormData = generateFormationFormData(
+        {
+          canCreateLimitedPartner: true,
+          canGetDistribution: true,
+          canMakeDistribution: true,
+        },
+        { legalStructureId }
+      );
+
+      expect(validatedFieldsForUser(formationFormData)).toContain("createLimitedPartnerTerms");
+      expect(validatedFieldsForUser(formationFormData)).toContain("getDistributionTerms");
+      expect(validatedFieldsForUser(formationFormData)).toContain("makeDistributionTerms");
+    });
+  });
+
+  describe("nonprofit", () => {
+    const legalStructureId = "nonprofit";
+
+    it("isVeteranNonprofit and Provisions radio questions does not have a response", () => {
```

Comment:
[pebble] This and the following tests:
1. Our pattern is to write these test names as sentences
2. I think `isVeteranNonprofit` should be removed from this test name, it's misleading because its value does not influence the outcome of the expectations.
3. Should be `hasNonprofitBoardMembers` radio questions not `Provisions` right?

Reply:
Done